homeassistant/components/recorder/util.py

```diff
@@ -407,6 +408,22 @@ def _fail_unsupported_version(
     raise UnsupportedDialect
 
 
+def _warn_deprecated_version(
+    server_version: str, dialect_name: str, minimum_version: str
+) -> None:
+    """Warn about upcoming unsupported database version."""
+    _LOGGER.warning(
+        (
+            "Support for version %s of %s is ending; the minimum supported version is %s. "
+            "Starting with Home Assistant 2024.2 this prevents the recorder from "
```

Comment:
I think we should make the HA version configurable, so we could use this function in the future again

Reply:
I'm going to refactor this to create an issue instead.  This was PR was intended as a starting point to give a view on the timeline for the arch PR